File: cases.erl
-module(cases).
-export([insert/2,beach/1]).

insert(X,[]) ->
  [X];
insert(X,Set) ->
  case lists:member(X,Set) of
    true -> Set;
    false -> [X|Set]
  end.

beach(Temprature) ->
  case Temprature of
    { celsius, N } when N >= 20, N =< 45 ->
      'favorable';
    { kelvin, N } when N >= 294, N =< 318 ->
      'scientifically favorable';
    { fahrenheit, N } when N >= 68, N =< 113 ->
      'favorable in the US';
    _ ->
      'avoid beach'
  end.

File: cases.ex
defmodule Cases do
  def insert(x,[]) do
    [x]
  end

  def insert(x,set) do
    case List.member(x,set) do
      true ->
        set
      false ->
        [x|set]
    end
  end

  def beach(temprature) do
    case temprature do
      { :celsius, n } when n >=2 and n <= 45 ->
        "favorable"
      { :kelvin, n } when n >= 294 and n <= 318 ->
        "scientifically favorable"
      { :fahrenheit, n } when n >= 68 and n <= 113 ->
        "favorable in the US"
      _ ->
        "avoid the beach"
    end
  end
end
